installation/src/Helper/DatabaseHelper.php

```diff
@@ -333,7 +333,7 @@ public static function checkRemoteDbHost($options)
     {
         // Security check for remote db hosts: Check env var if disabled. Also disable in CLI.
         $shouldCheckLocalhost = getenv('JOOMLA_INSTALLATION_DISABLE_LOCALHOST_CHECK') !== '1'
-            || php_sapi_name() == 'cli';
+            || php_sapi_name() !== 'cli';
```

Comment:
```suggestion
            && php_sapi_name() !== 'cli';
```
Localhost should be checked if the check is not disabled and the app is not CLI.

Reply:
I asked @SniperSister regarding this and we probably want to find a different solution. Don't have one yet...

Reply:
This doesn't change the fact that your logic is wrong and breaks the check for non-cli apps and my logic is right.

Reply:
It needs not much understanding of boolean algebra to see that an "or" is wrong here.

Reply:
So as long as your PR contains that code you should apply my suggestion.

Reply:
I already did. I also did not deny that the logic was wrong, but when I know that it will be "fundamentally" changed, I don't have to do 20 unnecessary commits.